### PR TITLE
ORC-363: Enable zstd for java writer/reader

### PR DESCRIPTION
**Pull request by wgtmac**
Use org.apache.hadoop.io.compress.ZStandardCodec in hadoop 2.9.0

### CODE REVIEW

File: java/pom.xml [hunk 1]
```diff
@@ -68,8 +68,8 @@
     <test.tmp.dir>${project.build.directory}/testing-tmp</test.tmp.dir>
     <example.dir>${project.basedir}/../../examples</example.dir>
 
-    <min.hadoop.version>2.2.0</min.hadoop.version>
-    <hadoop.version>2.7.3</hadoop.version>
+    <min.hadoop.version>2.9.0</min.hadoop.version>
```
[the remark this answers is not in the document]
Comment (wgtmac) [reply java/pom.xml]: That makes sense. Will do. Thanks!

**Comment by wgtmac** [comment]
@omalley I have changed this PR per your request. Please take a look again when you have time. Thanks!

**Comment by wgtmac** [comment]
@omalley Can you review this again when you have time? Thanks!

**Comment by wgtmac** [comment]
As several months passed by, it is time to bring it up again with some updates. Now I am also not in favor of this PR due to:
- It depends on org.apache.hadoop.io.compress.ZStandardCodec API and requires Hadoop version to be at least 2.9. It will be painful for some users to update Hadoop version.
- It requires zstd native library to be installed on cluster. However, pre-built Hadoop distributions not built with zstd by default. Though I managed to build Hadoop with zstd, it involved with a lot of workarounds and tricks including installing some dependencies and building specific versions of protoc and openssl. It is painful for users to do this.
- We cannot have E2E tests in this PR.
- The implementation uses JNI under the hood and some users are worried about its performance.

It seems to me that the only blocker is a pure Java implementation of zstd. Currently aircompressor has already supported zstd decompressor and its compressor is working in process: https://github.com/airlift/aircompressor/pull/93. I would wait until it is officially released and update this PR accordingly.

**Comment by jack-pappas** [comment]
@wgtmac @omalley FYI, the zstd support was merged in airlift/aircompressor earlier this month if you want to utilize it here.

**Comment by wgtmac** [comment]
@jack-pappas Thanks for notifying us. I will work on it shortly.

**Pull request closed without merging**